sdk/communication/communication-identity/test/public/utils/recordedClient.ts

```diff
@@ -48,107 +35,91 @@ const replaceableVariables: { [k: string]: string } = {
+): Promise<RecordedClient<CommunicationIdentityClient>> {
+  const recorder = await createRecorder(context.currentTest);
+
+  const client = new CommunicationIdentityClient(
+    env.COMMUNICATION_LIVETEST_DYNAMIC_CONNECTION_STRING ?? "",
+    recorder.configureClientOptions({})
+  );
 
   // casting is a workaround to enable min-max testing
   return {
-    client: new CommunicationIdentityClient(env.COMMUNICATION_LIVETEST_DYNAMIC_CONNECTION_STRING, {
-      httpClient: createTestHttpClient(),
-    } as CommunicationIdentityClientOptions),
+    client,
     recorder,
   };
 }
 
-export function createRecordedCommunicationIdentityClientWithToken(
+export async function createRecordedCommunicationIdentityClientWithToken(
   context: Context
-): RecordedClient<CommunicationIdentityClient> {
-  const recorder = record(context, environmentSetup);
+): Promise<RecordedClient<CommunicationIdentityClient>> {
+  const recorder = await createRecorder(context.currentTest);
+
   let credential: TokenCredential;
   const endpoint = parseConnectionString(
-    env.COMMUNICATION_LIVETEST_DYNAMIC_CONNECTION_STRING
+    env.COMMUNICATION_LIVETEST_DYNAMIC_CONNECTION_STRING ?? ""
   ).endpoint;
   if (isPlaybackMode()) {
     credential = {
-      getToken: async (_scopes) => {
+      getToken: async (_scopes: any) => {
         return { token: "testToken", expiresOnTimestamp: 11111 };
       },
     };
-
-    // casting is a workaround to enable min-max testing
-    return {
-      client: new CommunicationIdentityClient(endpoint, credential, {
-        httpClient: createTestHttpClient(),
-      } as CommunicationIdentityClientOptions),
-      recorder,
-    };
-  }
-
-  if (isNode) {
-    credential = new DefaultAzureCredential();
   } else {
-    credential = new ClientSecretCredential(
-      env.AZURE_TENANT_ID,
-      env.AZURE_CLIENT_ID,
-      env.AZURE_CLIENT_SECRET
-    );
+    credential = createTestCredential();
   }
 
-  // casting is a workaround to enable min-max testing
-  return {
-    client: new CommunicationIdentityClient(endpoint, credential, {
-      httpClient: createTestHttpClient(),
-    } as CommunicationIdentityClientOptions),
-    recorder,
-  };
-}
-
-function createTestHttpClient(): HttpClient {
-  const customHttpClient = new DefaultHttpClient();
-
-  const originalSendRequest = customHttpClient.sendRequest;
-  customHttpClient.sendRequest = async function (
-    httpRequest: WebResourceLike
-  ): Promise<HttpOperationResponse> {
-    const requestResponse = await originalSendRequest.apply(this, [httpRequest]);
-
-    console.log(
-      `MS-CV header for request: ${httpRequest.url} (${
-        requestResponse.status
-      } - ${requestResponse.headers.get("ms-cv")})`
-    );
+  const client = new CommunicationIdentityClient(
+    endpoint,
+    credential,
+    recorder.configureClientOptions({})
+  );
 
-    return requestResponse;
-  };
-
-  return customHttpClient;
+  // casting is a workaround to enable min-max testing
```

Comment:
nit: this appears to be an outdated comment

Reply:
thanks! removed